### PR TITLE
Feature/refine ogs

### PR DESCRIPTION
**Pull request by tkmct**

### CODE REVIEW

File: contracts/Dispute/CheckpointDispute.sol [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        require(
+            _inputs.length == 1,
+            "inputs length does not match. expected 1"
+        );
+        types.Property memory property = createClaimProperty(_inputs[0]);
+        disputeManager.settleGame(property);
+    }
+
+    // create checkpoint claim passed to dispute manager
+    // TODO: do we need this in property?
```
Comment (tkmct) [inline contracts/Dispute/CheckpointDispute.sol:284]: do not need inclusionProof

File: contracts/Dispute/CheckpointDispute.sol [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        types.Property memory challengeProperty = createChallengeProperty(
+            _challengeInputs[0]
+        );
+
+        require(
+            disputeManager.isChallengeOf(property, challengeProperty),
+            "Invalid challenge"
+        );
+
+        // TODO: is it okay to use CompiledPredicate interface here?
```
Comment (tkmct) [inline contracts/Dispute/CheckpointDispute.sol:255]: ok

File: contracts/Dispute/CheckpointDispute.sol [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            stateUpdate.depositContractAddress ==
+                challengeStateUpdate.depositContractAddress,
+            "DepositContractAddress is invalid"
+        );
+        require(
+            stateUpdate.blockNumber > challengeStateUpdate.blockNumber,
+            "BlockNumber must be smaller than challenged state"
+        );
+        require(
+            isSubrange(challengeStateUpdate.range, stateUpdate.range), // TODO: is this right?
```
Comment (tkmct) [inline contracts/Dispute/CheckpointDispute.sol:172]: ok

File: test/Dispute/CheckpointDispute.test.ts [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+          checkpointDispute.claim(inputs, witness, {
+            gasLimit: 1200000
+          })
+        ).to.be.reverted
+      })
+    })
+  })
+
+  describe('challenge', () => {
+    describe('succeed to challeng echeckpoint', () => {
```
Comment (syuhei176) [inline test/Dispute/CheckpointDispute.test.ts:283]: typo;;) checkpoint

File: test/Dispute/CheckpointDispute.test.ts [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+        await commitment.submitRoot(nextBlockNumber, firstBlockInfo.root)
+
+        const secondBlockInfo = prepareBlock(BOB_ADDRESS, nextBlockNumber + 1)
+        await commitment.submitRoot(nextBlockNumber + 1, secondBlockInfo.root)
+
+        const inputs = [encodeStructable(secondBlockInfo.stateUpdate.property)]
+        const witness = [encodeStructable(secondBlockInfo.inclusionProof)]
+
+        await checkpointDispute.claim(inputs, witness, {
+          gasLimit: 1200000
```
Comment (syuhei176) [inline test/Dispute/CheckpointDispute.test.ts:298]: Can you make it smaller?

File: test/Dispute/CheckpointDispute.test.ts [hunk 6]
```diff
@@ -0,0 +1,756 @@
...
+        const currentBlockNumber = await commitment.currentBlock()
+        const nextBlockNumber = currentBlockNumber + 1
+
+        const stateUpdate = ownershipStateUpdate(
+          Address.from(ALICE_ADDRESS),
+          nextBlockNumber,
+          0,
+          5
+        )
+        const { root, falsyInclusionProof } = generateTree(stateUpdate)
```
Comment (syuhei176) [inline test/Dispute/CheckpointDispute.test.ts:267]: Could you explain how did you make false inclusion proof?

File: test/Dispute/CheckpointDispute.test.ts (hunk 4 above)
Comment (tkmct) [reply test/Dispute/CheckpointDispute.test.ts:283]: Thanks!

File: test/Dispute/CheckpointDispute.test.ts (hunk 5 above)
Comment (tkmct) [reply test/Dispute/CheckpointDispute.test.ts:298]: Sure.

File: test/Dispute/CheckpointDispute.test.ts (hunk 6 above)
Comment (tkmct) [reply test/Dispute/CheckpointDispute.test.ts:267]: it just returns inclusionProof of same tree but different index.

**Pull request merged**